scipy/sparse/linalg/_matfuncs.py

```diff
@@ -861,3 +862,47 @@ def _ell(A, m):
     log2_alpha_div_u = np.log2(alpha/u)
     value = int(np.ceil(log2_alpha_div_u / (2 * m)))
     return max(value, 0)
+
+def matrix_power(A, power, structure=None):
+    """
```

Comment:
A couple docstring issues:

* As @jjerphan already noted, `structure` must be described in the "Parameters" section.  If there is nothing implemented yet, then remove the parameter.
* Add an "Examples" section (see https://github.com/scipy/scipy/issues/7168).
* Add a "Notes" section with the appropriate `versionadded` markup.

For comparison, see the `expm` docstring in this file.  A "Notes" section with just the `versionadded` markup is fine.

Reply:
Thanks! Have done. "Structure" was a holdover from using the `MatrixPowerOperator` implementation, and it has been removed. I also refer now to the idea of your stackoverflow comment in the notes... I think it's pretty important to disclose this to the user.